src/engine/renderer/glsl_source/reliefMapping_fp.glsl

```diff
@@ -199,6 +199,27 @@ void computeDLights( vec3 P, vec3 N, vec3 I, vec4 diffuse, vec4 specular,
 #endif
 }
 
+vec3 ComputeNormalInTangentSpace(sampler2D normalMap, vec2 texNormal)
+{
+	vec3 N = texture2D(u_NormalMap, texNormal.st).xyw;
+	N.x *= N.z;
+	N.xy = 2.0 * N.xy - 1.0;
+	N.z = sqrt(1.0 - dot(N.xy, N.xy));
+
+#if defined(r_NormalScale)
+	N.z *= r_NormalScale;
+	normalize(N);
```

Comment:
does this line have any effect?

Reply:
the `normalize` one? I really don't know

Reply:
I don't think it does. If you don't know, where are you getting the code from?

Reply:
copy pasted from our code, this is a refactor, some of the normal code in other files had this.

Reply:
Let's just delete this `r_NormalScale` stuff. The code does not make sense: (1) Why would you want to multiply just one component of the vector by some value? (2) According to any docs/examples I find, `normalize` is supposed to be used by the return value, not an in-place modification.

Nothing will be lost as it would only have an effect if you are playing with cvars.

Reply:
The `z` component is the vector that goes toward the viewer, that may increase the perceived depth and contrast the shadows. That may be useful as a per-shader keyword, and it is a per-shader keyword. The mistake there is that i's named `r_normalScale` and not `u_normalScale`.

I noticed that numbers under 1 darken the shadows, numbers over 1 soften. I also noticed the `normalize` line seems to not affect the rendering and may be removed. If you want to try it with `r_normalScale`, you need `vid_restart` to see the changes.

With `r_normalScale .1`: 

[![normalscale](https://dl.illwieckz.net/b/daemon/features/normalscale/unvanquished_2019-03-18_193327_000.jpg)](https://dl.illwieckz.net/b/daemon/features/normalscale/unvanquished_2019-03-18_193327_000.jpg)

[![normalscale](https://dl.illwieckz.net/b/daemon/features/normalscale/unvanquished_2019-03-18_193453_000.jpg)](https://dl.illwieckz.net/b/daemon/features/normalscale/unvanquished_2019-03-18_193453_000.jpg)

[![normalscale](https://dl.illwieckz.net/b/daemon/features/normalscale/unvanquished_2019-03-18_193525_000.jpg)](https://dl.illwieckz.net/b/daemon/features/normalscale/unvanquished_2019-03-18_193525_000.jpg)

I removed the useless `normalize` call.

I was already thinking about revamping that part to be sure the shader keyword value is taken in account _and_ in any case, but I'm not sure to do it within that PR. It's planed in any way.